src/transaction/log_storage.hpp

```diff
@@ -81,10 +83,27 @@ typedef struct log_page LOG_PAGE;
 struct log_page
 {
   /* The log page */
+  bool operator== (const log_page &other);
+
   LOG_HDRPAGE hdr;
   char area[1];
 };
 
+class log_page_wrapper
+{
+  public:
+    explicit log_page_wrapper (const char *buffer);
+    ~log_page_wrapper ();
+
+    bool operator== (const log_page_wrapper &other);
+    bool operator== (const LOG_PAGE &other);
+    const LOG_HDRPAGE &get_header () const;
+
+  private:
+    char *m_buffer;
+    LOG_PAGE *m_log_page;
```

Comment:
Tried this change and it gave a really tough time to implement the == operator and also since I'm not really sure that the start address of a std::string is the same as it's char array...the memcpy might be more difficult to implement. .c_str() doesn't work because it's a const.